### PR TITLE
add Path extension assertion

### PR DESCRIPTION
**Pull request by lelloman**
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt [hunk 1]
```diff
@@ -11,13 +11,15 @@ import ch.tutteli.atrium.translations.DescriptionCharSequenceAssertion.ENDS_NOT_
 import ch.tutteli.atrium.translations.DescriptionPathAssertion.*
 import ch.tutteli.niok.exists
 import ch.tutteli.niok.fileNameWithoutExtension
+import ch.tutteli.kbox.isNotNullAndNotEmpty
```
Comment (lelloman) [inline domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt]: This is a leftover, will remove on next push.

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt [hunk 2]
```diff
@@ -147,3 +147,24 @@ val <T : Path> Expect<T>.parent get(): Expect<Path> = ExpectImpl.path.parent(thi
...
+ *
+ * @return The newly created [Expect].
+ *
+ * @since 0.9.0
+ */
+val <T : Path> Expect<T>.extension get(): Expect<String> =
+    ExpectImpl.path.extension(this).getExpectOfFeature()
+
+/**
+ * Expects that this [Path] has a given extension.
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt]: please copy KDOC from `fileName` with assertionCreator above and adapt

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt [hunk 3]
```diff
@@ -147,3 +147,24 @@ val <T : Path> Expect<T>.parent get(): Expect<Path> = ExpectImpl.path.parent(thi
  */
 fun <T : Path> Expect<T>.parent(assertionCreator: Expect<Path>.() -> Unit): Expect<T> =
     ExpectImpl.path.parent(this).addToInitial(assertionCreator)
+
+/**
+ * Creates an [Expect] for the extension of this Path so that further fluent calls are assertions about it.
+ *
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt]: please copy KDOC from `fileName` above and adapt

Comment (lelloman) [reply apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt]: done

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt (hunk 2 above)
Comment (lelloman) [reply apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt]: done

**Comment by robstoll** [comment]
LGTM, thanks for the contribution :+1:

**Comment by lelloman** [comment]
No problem, do you think I can pick this one up https://github.com/robstoll/atrium/issues/127? I would still do it at a relaxed pace though.

**Comment by robstoll** [comment]
yes you can, please write a comment there do that I can assign you to the issue

**Pull request merged**